subprojects/core/src/integTest/groovy/org/gradle/api/invocation/GradleLifecycleIntegrationTest.groovy

```diff
@@ -130,18 +133,21 @@ class GradleLifecycleIntegrationTest extends AbstractIntegrationSpec {
                     public void apply(Settings target) {
                         // Expose dsl to the user, the value will be isolated only after settings has been fully evaluated
                         final Dsl dsl = target.getExtensions().create("dsl", Dsl.class);
-                        target.getGradle().getLifecycle().beforeProject(project -> {
+                        target.getGradle().getLifecycle().$lifecycleApi(project -> {
                             project.getTasks().register("test", TestTask.class, task -> {
                                 task.getTaskParameter().set(dsl.getParameter());
                             });
                         });
                     }
                 }
-            '''
+            """
         }
 
         expect:
         configuredTaskRunsCorrectly()
+
+        where:
+        lifecycleApi << ["allprojects", "beforeProject"]
     }
 
     def "lifecycle beforeProject/afterProject run around project evaluation"() {
```

Comment:
💭 We should update this test to include `lifecycle.allprojects`. This test demonstrates the relation between all of these callback mechanisms